test/DatabaseIntegrationTest/Tools/Commands/Users/AddUserCommand.cpp

```diff
@@ -0,0 +1,34 @@
+#include "stdafx.h"
+#include "AddUserCommand.h"
+
+#include "DatabaseIntegrationTest/Tools/Core.h"
+
+#include "SeedCppCore/Model/User.h"
+#include "SeedCppCore/Model/UserMgr.h"
+#include "SeedCppCore/Services/ServicesMgr.h"
+#include "SeedCppCore/Services/Model/ModelServicesMgr.h"
+#include "SeedCppCore/Services/Model/IUserModelService.h"
+
+
+namespace seed_cpp { namespace db_test {
+
+	AddUserCommand::AddUserCommand(const model::User& user)
+		:m_user(std::make_unique<model::User>(user))
+	{
+	}
+
+	AddUserCommand::~AddUserCommand() = default;
+
+	void AddUserCommand::execute(Core& core)
+	{
+		auto& service = core.getServicesMgr().getModelServicesMgr().getUserModelService();
+		model::UserMgr::UniqueLock lock(service.getEntityMgr());
```

Comment:
You have a method in the ModelService to generate this lock.

Reply:
Agree. Updated to use it.